File: java/PartTimeEmployee.java
import java.util.Date;

public abstract class PartTimeEmployee extends Employee {
    private double hourlyRate;
    private int hoursPerWeek;

    public PartTimeEmployee(String id, String firstName, String lastName, Date dob,
            String position, String department, double hourlyRate, int hours) throws InvalidDataException {
        super(id, firstName, lastName, dob, position, department);
        this.hourlyRate = hourlyRate;
        this.hoursPerWeek = hours;
    }

    @Override
    public double getAnnualSalary() {
        return hourlyRate * hoursPerWeek * 52;
    }

    @Override
    public void reportToManager() {
        if (manager != null) {
            System.out.println(getFullName() + " (Part-Time) reports to manager " + manager.getFullName());
        } else {
            System.out.println(getFullName() + " (Part-Time) has no assigned manager.");
        }
    }
}

File: csharp/PartTimeEmployee.cs
using System;

[Serializable]
public abstract class PartTimeEmployee : Employee
{
    public double HourlyRate { get; set; }
    public int HoursPerWeek { get; set; }

    protected PartTimeEmployee() { }

    protected PartTimeEmployee(string id, string firstName, string lastName, DateTime dob,
        string position, string department, double hourlyRate, int hoursPerWeek)
        : base(id, firstName, lastName, dob, position, department)
    {
        HourlyRate = hourlyRate;
        HoursPerWeek = hoursPerWeek;
    }

    public override double GetAnnualSalary() => HourlyRate * HoursPerWeek * 52;

    public override void ReportToManager()
    {
        if (Manager != null)
            Console.WriteLine($"{GetFullName()} (Part-Time) reports to manager {Manager.GetFullName()}");
        else
            Console.WriteLine($"{GetFullName()} (Part-Time) has no assigned manager.");
    }
}
